### PR TITLE
updating task and corresponding unit tests

### PR DESCRIPTION
**Pull request by dhrubo-os**
### Description
[updating task and corresponding unit tests]

### Related Issues
Resolves #[Issue number to be closed when this PR is merged]

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).
- [ ] Commits are signed per the DCO using `--signoff`.
- [ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/ml-commons/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java [hunk 1]
```diff
@@ -333,30 +337,33 @@ public void updateMLTask(
...
-                    actionListener.onFailure(e);
-                }
+
+                UpdateDataObjectRequest updateDataObjectRequest = UpdateDataObjectRequest
+                    .builder()
+                    .index(ML_TASK_INDEX)
+                    .id(taskId)
+                    .tenantId(tenantId)
+                    .dataObject(updatedContent)
+                    .retryOnConflict(3)
```
Comment (dbwiddis) [inline plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java:358]: In the previous, this retry is conditional on the existence of a STATE_FIELD. Should that be preserved?

File: plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java [hunk 2]
```diff
@@ -333,30 +337,33 @@ public void updateMLTask(
...
+                    .build();
+
+                sdkClient
+                    .updateDataObjectAsync(updateDataObjectRequest, client.threadPool().executor(GENERAL_THREAD_POOL))
+                    .whenComplete((r, throwable) -> {
+                        semaphore.release();
```
Comment (dbwiddis) [inline plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java:364]: Previous code null-checks this semaphore. Is that needed here?

File: plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java (hunk 1 above)
Comment (dhrubo-os) [reply plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java]: Yeah we should. Thanks for catching this.

File: plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java (hunk 2 above)
Comment (dhrubo-os) [reply plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java]: Yeah I added it back. Thanks.

**Review by dbwiddis — APPROVED** [review]
LGTM.  Started leaving some comments but see they're handled (pun intended).

**Pull request merged**